datastore/tiered_datastore.nim

```diff
@@ -26,57 +28,94 @@ proc stores*(self: TieredDatastore): seq[Datastore] =
 
 method contains*(
   self: TieredDatastore,
-  key: Key): ?!bool {.locks: "unknown".} =
+  key: Key): Future[?!bool] {.async, locks: "unknown".} =
 
   var
     exists = false
```

Comment:
I think it wiull be shorter and probably cleaner without the `exists` variable

Reply:
The natural alternative would be:
```
...
if containsRes.get: break

return success containsRes.get
```

For that to work I'd need to declare `var containsRes` outside of the scope of the `for` loop, i.e. in the same place `var exists` is currently declared.  Altogether that doesn't seem much cleaner than the approach I took with `var exists` but I can make the change if that's what is preferred.

Reply:
I mean smth like that, the main problem is how to make it clear that we return true on the first containsRes==true:
```
for store in self.stores:
    let containsRes = await store.contains(key)
    if containsRes.isErr: return containsRes
    if containsRes.get == true: return success true

  return success false
```
It's 5 non-empty lines instead of 7

Reply:
Change made.